satellite/nodestats/endpoint.go

```diff
@@ -45,12 +47,11 @@ func (e *Endpoint) GetStats(ctx context.Context, req *pb.GetStatsRequest) (_ *pb
 
 	peer, err := identity.PeerIdentityFromContext(ctx)
 	if err != nil {
-		return nil, NodeStatsEndpointErr.Wrap(err)
+		return nil, status.Error(codes.Unauthenticated, err.Error())
 	}
-
 	node, err := e.overlay.Get(ctx, peer.ID)
 	if err != nil {
-		return nil, NodeStatsEndpointErr.Wrap(err)
+		return nil, status.Error(codes.Unauthenticated, err.Error())
```

Comment:
This error could be originated because of an error with the database (connection failure, etc.), in such case, the error code should be `Internal`, for the rest of the cases this code is good.

Reply:
Fixed. Also realized that the error should be `PermissionDenied` as well.